app/helpers/course/assessment/submission/submissions_workflow_state_helper.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+module Course::Assessment::Submission::SubmissionsWorkflowStateHelper
+  # Returns a hash of course_user to submission for searching by course_user
+  #
+  # @return [Hash]
+  def submissions_hash
+    @submissions_hash ||= @submissions.map { |s| [s.course_user, s] }.to_h
```

Comment:
I think we might not need to load the course_user here, can use `s.course_user_id` as the key and it's probably faster in hash lookup as well.